httpcore/_types.py

```diff
@@ -2,10 +2,10 @@
 Type definitions for type checking purposes.
 """
 
-from typing import Dict, List, Optional, Tuple, Union
+from typing import List, Mapping, Optional, Tuple, Union
 
 StrOrBytes = Union[str, bytes]
 Origin = Tuple[bytes, bytes, int]
 URL = Tuple[bytes, bytes, Optional[int], bytes]
 Headers = List[Tuple[bytes, bytes]]
```

Comment:
I see this is a `List` and not a `Sequence` but the value type doesn't vary so that's ok. Still, it might be nice to have some standardisation on when to use `List` vs `Sequence`/`Dict` vs `Mapping` instead of trying to fix these every time somebody's mypy complains.

Reply:
I guess the standard is what you've just mentioned - invariant type is OK if inner types don't vary, but covariant is required when they do.

An option to catch these more early on would be to systematically define these arguments as separate variables first in tests, since this is how these MyPy errors fire off usually... It's a bit tedious but we could see what a PR for this looks like...